pages/api/calendar.js

```diff
@@ -0,0 +1,33 @@
+import { DateTime } from 'luxon';
+
+import { getTalks } from '../../lib/airtable';
+
+const currentDate = DateTime.local();
+
+const removeOldTalks = talk => new Date(talk.date) > new Date();
+
+const futureTalks = month => ({
+    date: currentDate.set({ month }).toISO(),
+});
+
+const completeNextYearTalks = sizeCurrentMonth =>
+    Array.from(
+        {
+            length: 12 - sizeCurrentMonth,
+        },
+        (_, i) => futureTalks((i + new Date().getMonth() + 1 + sizeCurrentMonth) % 12),
+    );
+
+export default async (req, res) => {
+    try {
+        const talks = await getTalks();
+
+        const futureTalks = talks.filter(removeOldTalks);
+        const calendar = [...futureTalks, ...completeNextYearTalks(futureTalks.length)];
+        const notPlanned = talks.filter(talk => !talk.date);
+
+        return res.status(200).json({ data: { calendar, notPlanned } });
```

Comment:
Why encapsulate the return data in a sub property and not return it directly?

Reply:
It's more generic, you know you have data object with contains "data" and in case of error you have error object.

I can remove if you want :)

Reply:
It's more generic, you know you have data object with contains "data" and in case of error you have error object.

I can remove if you want :)

Reply:
I'd say remove it, in case of error, the status of the response should be enough to control the code flow on the calling side. It will also simplify the deconstruction of the response on the client side